src/main/java/org/eclipse/basyx/vab/modelprovider/VABPathTools.java

```diff
@@ -218,8 +218,17 @@ public static String append(String path, String element) {
 		if (path == null || element == null) {
 			return null;
 		}
-
-		if (path.lastIndexOf(SEPERATOR) == path.length() - 1) {
+		if (element.length() == 0) {
+			return path;
+		}
+		boolean pathHasLastSeparator = path.lastIndexOf(SEPERATOR) == path.length() - 1;
```

Comment:
Please rename these variables to be more clear.
E. g.: ``pathHasSeparatorAsLastChar``. It took me a few minutes to understand what is done here.